packages/elevenlabs/src/elevenlabs-transcription-model.ts

```diff
@@ -0,0 +1,181 @@
+    private readonly config: ElevenLabsTranscriptionModelConfig,
+  ) {}
+
+  private getArgs({
+    audio,
+    mediaType,
+    providerOptions,
+  }: Parameters<TranscriptionModelV1['doGenerate']>[0]) {
+    const warnings: TranscriptionModelV1CallWarning[] = [];
+
+    // Parse provider options
+    const elevenlabsOptions = parseProviderOptions({
+      provider: 'elevenlabs',
+      providerOptions,
+      schema: elevenLabsProviderOptionsSchema,
+    });
+
+    // Create form data with base fields
+    const formData = new FormData();
+    const blob =
+      audio instanceof Uint8Array
+        ? new Blob([audio])
+        : new Blob([convertBase64ToUint8Array(audio)]);
+
+    formData.append('model_id', this.modelId);
+    formData.append('file', new File([blob], 'audio', { type: mediaType }));
+    formData.append('diarize', 'true');
+
+    // Add provider-specific options
+    if (elevenlabsOptions) {
+      const transcriptionModelOptions: ElevenLabsTranscriptionAPITypes = {
+        language_code: elevenlabsOptions.languageCode ?? undefined,
+        tag_audio_events: elevenlabsOptions.tagAudioEvents ?? undefined,
+        num_speakers: elevenlabsOptions.numSpeakers ?? undefined,
+        timestamps_granularity:
+          elevenlabsOptions.timestampsGranularity ?? undefined,
+        file_format: elevenlabsOptions.file_format ?? undefined,
+      };
+
+      if (typeof elevenlabsOptions.diarize === 'boolean') {
+        formData.append('diarize', String(elevenlabsOptions.diarize));
+      }
+
+      for (const key in transcriptionModelOptions) {
+        const value =
+          transcriptionModelOptions[
+            key as keyof ElevenLabsTranscriptionAPITypes
+          ];
+        if (value !== undefined) {
+          formData.append(key, String(value));
+        }
+      }
+    }
+
+    return {
+      formData,
+      warnings,
+    };
+  }
+
+  async doGenerate(
+    options: Parameters<TranscriptionModelV1['doGenerate']>[0],
+  ): Promise<Awaited<ReturnType<TranscriptionModelV1['doGenerate']>>> {
+    const currentDate = this.config._internal?.currentDate?.() ?? new Date();
+    const { formData, warnings } = this.getArgs(options);
+
+    const {
+      value: response,
+      responseHeaders,
+      rawValue: rawResponse,
+    } = await postFormDataToApi({
+      url: this.config.url({
+        path: '/v1/speech-to-text',
+        modelId: this.modelId,
+      }),
+      headers: combineHeaders(this.config.headers(), options.headers),
+      formData,
+      failedResponseHandler: elevenlabsFailedResponseHandler,
+      successfulResponseHandler: createJsonResponseHandler(
+        elevenlabsTranscriptionResponseSchema,
+      ),
+      abortSignal: options.abortSignal,
+      fetch: this.config.fetch,
+    });
+
+    return {
+      text: response.text,
+      segments:
+        response.words?.map(word => ({
+          text: word.text,
+          startSecond: word.start ?? 0,
+          endSecond: word.end ?? 0,
```

Comment:
no need to change here but could those default lead to repercussions in practice? should we support `null` as `startSecond`/`endSecond`?

Reply:
Good call, I'll add that to my todo list